internal/querynode/shard_cluster.go

```diff
@@ -128,12 +128,15 @@ type ShardCluster struct {
 	segmentDetector ShardSegmentDetector
 	nodeBuilder     ShardNodeBuilder
 
-	mut            sync.RWMutex
-	leader         *shardNode                     // shard leader node instance
-	nodes          map[int64]*shardNode           // online nodes
-	segments       SegmentsStatus                 // shard segments
-	versions       map[int64]*ShardClusterVersion // version id to version map
-	currentVersion *ShardClusterVersion           // current serving segment state version
+	mut      sync.RWMutex
+	leader   *shardNode           // shard leader node instance
+	nodes    map[int64]*shardNode // online nodes
+	segments SegmentsStatus       // shard segments
+
+	mutVersion sync.RWMutex
+	//versions       map[int64]*ShardClusterVersion // version id to version map
+	versions       sync.Map             // version id to version
+	currentVersion *ShardClusterVersion // current serving segment state version
```

Comment:
since version is already sync.map. what about using Atomic for current value?

Reply:
mutVersion needs to block get version's segment allocation